### PR TITLE
Fixes Extension window doesn't popup on request

### LINKED ISSUE #13590: [Bug]: Extension window doesn't popup on request
### Describe the bug

Metamask 10.9.2
Chrome 98.0.4758.80 (Official Build) (arm64)

If metamask is not unlocked yet (fresh browser start), the popup window doesn't show.

If the extension is pinned, I can see the "notification" number appearing, other than that there's no feedback. And if the extension is not pinned, there's no way for the user to tell there's an action expected from him.

Request promise is forever pending, and if the page is reloaded the new request errors out with:
`MetaMask - RPC Error: Already processing eth_requestAccounts. Please wait. {code: -32002, message: 'Already processing eth_requestAccounts. Please wait.'}`

[This issue](https://github.com/MetaMask/metamask-extension/issues/4006) describes similar problem, but it was closed a year ago.

### Steps to reproduce

1. Relaunch chrome so the extension locks
2. Go to [metamask docs site example section](https://docs.metamask.io/guide/getting-started.html#connecting-to-metamask)
3. Click on the 'enable ethereum' button, which is supposed to activate the popup
4. Nothing happens

### Error messages or log output

_No response_

### Version

10.9.2

### Build type

_No response_

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by segun**
Fixes: #13590 

Explanation:  

Manual testing steps:  
  - 
  - 
  -

### CODE REVIEW

**Review by 0ex-d — COMMENTED** [review]
I think we can close this one

File: app/scripts/lib/rpc-method-middleware/handlers/request-accounts.js [hunk 1]
```diff
@@ -72,7 +72,7 @@ async function requestEthereumAccountsHandler(
     // lock state when they were received.
     try {
       locks.add(origin);
-      await getUnlockPromise();
+      await getUnlockPromise(true);
```
[the remark this answers is not in the document]
Comment (segun) [reply app/scripts/lib/rpc-method-middleware/handlers/request-accounts.js:75]: This is the definition of `getUnlockPromise`
[getUnlockPromise](https://github.com/MetaMask/metamask-extension/blob/3327c5c7979b7cf90a9ad0f29d5b7edcae4da9f0/app/scripts/controllers/app-state.js#L73)

`getUnlockPromise` calls `waitForUnlock`
[waitForUnlock](https://github.com/MetaMask/metamask-extension/blob/3327c5c7979b7cf90a9ad0f29d5b7edcae4da9f0/app/scripts/controllers/app-state.js#L92)

if nothing is passed (the bug) this check in `waitForUnlock` fails
[waitForUnlock](https://github.com/MetaMask/metamask-extension/blob/3327c5c7979b7cf90a9ad0f29d5b7edcae4da9f0/app/scripts/controllers/app-state.js#L95)
And the popup is not shown.

**Review by jpuri — APPROVED** [review]
🚀

**Review by NiranjanaBinoy — APPROVED** [review]

**Comment by danjm** [comment]
The removal of this line in particular is why this fix was needed https://github.com/MetaMask/metamask-extension/pull/12243/files#diff-dd58fd574b3f61106e9c9952e1c5457ab8ef7acb1949f50c44683c8a706cd6abL102

**Pull request merged**